bootstrap/004_labels.sql

```diff
@@ -263,114 +166,66 @@ END;
 
 CREATE OR REPLACE PROCEDURE ADMIN.DELETE_LABEL(name text)
     RETURNS text
-    LANGUAGE SQL
+    language python
+    runtime_version = "3.10"
+    handler = 'delete_label'
+    packages = ('snowflake-snowpark-python', 'pydantic')
+    imports = ('{{stage}}/python/crud.zip')
     EXECUTE AS OWNER
 AS
-BEGIN
-    if (:name is null) then
-      return 'Name must not be null.';
-    end if;
-
-    DELETE FROM internal.labels where name = :name;
-    CALL INTERNAL.UPDATE_LABEL_VIEW();
-    return 'done';
-END;
+$$
+from crud import delete_entity
+def delete_label(session, name):
+    return delete_entity(session, 'LABEL', name)
+$$;
 
+-- TODO Differentiate from ungrouped labels  by having "AND IS_DYNAMIC" in the delete clause
 CREATE OR REPLACE PROCEDURE ADMIN.DELETE_DYNAMIC_LABEL(name text)
     RETURNS text
-    LANGUAGE SQL
+    language python
+    runtime_version = "3.10"
+    handler = 'delete_label'
+    packages = ('snowflake-snowpark-python', 'pydantic')
+    imports = ('{{stage}}/python/crud.zip')
     EXECUTE AS OWNER
 AS
-BEGIN
-    if (:name is null) then
-      return 'Name must not be null.';
-    end if;
-
-    DELETE FROM internal.labels where group_name = :name and is_dynamic;
-    CALL INTERNAL.UPDATE_LABEL_VIEW();
-    return 'done';
-END;
+$$
+from crud import delete_entity
+def delete_label(session, name):
+    return delete_entity(session, 'LABEL', name)
+$$;
 
 CREATE OR REPLACE PROCEDURE ADMIN.UPDATE_LABEL(oldname text, name text, grp text, rank number, condition text, is_dynamic boolean)
```

Comment:
is it possible to get rid of this one? Do we need the is_dynamic still?

Reply:
Same with this one -- intentionally didn't collapse the overloaded functions. If it's good with you, would do that in a follow-on (likely today)

Reply:
foolow on is perfect